src/maintypes.jl

```diff
@@ -0,0 +1,48 @@
+"""
+    Manifold{F}
+
+A manifold type. The `Manifold` is used to dispatch to different functions on a manifold,
+usually as the first argument of the function. Examples are the [`exp`](@ref)onential and
+[`log`](@ref)arithmic maps as well as more general functions that are built on them like the
+[`geodesic`](@ref).
+
+The manifold is parametrized by an [`AbstractNumbers`](@ref) to distinguish for example
+real (ℝ) and complex (ℂ) manifolds.
+
+For subtypes the preferred order of parameters is: size and simple value parameters,
+followed by the [`AbstractNumbers`](@ref) `field`, followed by data type parameters,
+which might depend on the abstract number field type.
+"""
+abstract type Manifold{𝔽} end
```

Comment:
Maybe you could `@ref` the section in docs about implementing a new manifold?

Reply:
Interesting idea. My only point, where I am unsure, is, that if you call this in REPL, you would just see "How to implement a manifold"?

Reply:
No, that's not what I had in mind. I think that `@ref` could be added at the bottom of the docs, and not replace the text that's here.

Reply:
Yes, that‘s what I understood :) Still in the REPL this last (additional) line would not help that much since its not a link there?

Reply:
It would be helpful when browsing the online docs. Additionally we can have a URL in the docstring that would be helpful in REPL.

Reply:
Sure. I am not 100% a fan of that, since the URL would be static (e.g. also link to the stable tutorial when looking at the dev variant, but we could do that.

Reply:
We could also add it to the `interface.md` just before the docs block containing this type?

Reply:
Add what to `interface.md`? I'd just like to have something that would point people who read the docstring of `Manifold` type to `interface.md`.

Reply:
Ah, I though something that links to the tutorial, that could be done in interface. Hm, then I will think of something.